### DISCUSSION TITLE
Wyze Sensor Support

### DISCUSSION THREAD

**Discussion by faanskit** [SecKatie/ha-wyzeapi#166] [Q&A]
With [PR 5](https://github.com/JoshuaMulliken/wyzeapy/pull/5) in Wyzeapy, Wyze sensor (v1) implementation is super simple:

**binary_sensor.py**
```
43
         -   if device_type == DeviceTypes.CAMERA:
         +   if device_type == DeviceTypes.CAMERA or device_type == DeviceTypes.MOTION_SENSOR or device_type == DeviceTypes.CONTACT_SENSOR:
```
It would also make sense to rename `class WyzeCameraMotion()` to `class WyzeSensor()`, and `cameras` to `sensor` in `async_setup_entry()`

As an option, a new class can be introduced parallel to WyzeCameraMotion(). Or in fact, the following classes could be introduced:

- `class WyzeCameraMotion` : Binary Sensor: motion
- `class WyzeCameraSound` : Binary Sensor: sound
- `class WyzeCameraFire` : Binary Sensor: gas (#161) (EDIT) - hard to reverse engineer; need to set off alarm :-)
- `class WyzeCameraCo2` : Binary Sensor: smoke (#161) (EDIT) - hard to reverse engineer; need to set off alarm :-)
- `class WyzeContactSensor` : Binary Sensor: door (or window, or opening)
- `class WyzeMotionSensor` : Binary Sensor: motion

Contact and Motion sensors would benefit from exposing _RSSI_ and _Battery level_. Learnings can be made from [fork](https://github.com/jfarmer08/ha-wyzeapi) I think.

Opinions or proposed way forward?

**Comment by faanskit** [comment]
Why were Sensors removed in #129 ?

**Reply by BarrettPalmer** [reply]
I agree it would be nice to have the sensors info back. I currently am getting info from a homebridge plug-in. I would like to see the climate sensor info. Can you please add this back?

**Reply by SecKatie** [reply]
They were removed because it was hammering the Wyze servers and they threatened to shut down all of our access if we didn't reduce the load

**Comment by BarrettPalmer** [comment]
Thank you for the info! Really appreciate your work.
